docker-compose.override.yaml

```diff
@@ -0,0 +1,19 @@
+version: "3.7"
+
+services:
+  postgres:
+    image: kartoza/postgis:11.0-2.5
+    environment:
+      - POSTGRES_DB=opendatacube
+      - POSTGRES_PASSWORD=opendatacubepassword
+      - POSTGRES_USER=opendatacube
+    ports:
+      - 5432:5432
+    restart: always
+
+  sandbox:
+    environment:
+      - DB_HOSTNAME=postgres
```

Comment:
I would prefer a single environment variable here:

As describe here:
https://datacube-core.readthedocs.io/en/latest/ops/config.html#configuration-via-environment-variables
```yaml
- DATACUBE_DB_URL=postgresql://opendatacube:opendatacubepassword@postgres/opendatacube
```

If it is possible with docker compose I would use some form of indirection to make sure that both docker images use the same values for user/password/db

Reply:
It doesn't make anything easier having a single variable. I think the connection strings are a pain to parse, as a human. I know it is possible, but I don't think it's needed...

Reply:
@alexgleith  just FYI the other method is deprecated in `datacube` code.

Single variable method with url format might be extended in the future to include multiple database or include extra flags that are db backend specific when we finally do move to ODC 2.0, at this point I will simply remove support for multi-environment variable configuration altogether. So, rather than propagating this deprecated approach further I would very much prefer for this to be changed now and not later. There is also an issue of these variables being completely generic, no ODC specific name-spacing which makes it dangerous. I only put in support for these variables to help with deployment migration with an understanding that people will migrate to the new format. But since that's not happening at all, I might as well delete support for this method of configuration before 1.8 release.

Reply:
Right, well I didn't realise this method was deprecated.

How about I start an issue on the datacube-core github and we can work it out.

I like having multiple environment variables... but let's seek a consensus :-)

Reply:
This has always been a backward compatibility with old docker thing. `DATACUBE_DB_URL` format is what is officially documented as supported method for environment variable based config.